tests/sqllogictests/src/main.rs

```diff
@@ -0,0 +1,223 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs::ReadDir;
+use std::path::PathBuf;
+
+use clap::Parser;
+use client::ClickhouseHttpClient;
+use sqllogictest::DBOutput;
+use walkdir::DirEntry;
+use walkdir::WalkDir;
+
+use crate::arg::SqlLogicTestArgs;
+use crate::client::HttpClient;
+use crate::client::MysqlClient;
+use crate::error::DSqlLogicTestError;
+use crate::error::Result;
+use crate::util::find_specific_dir;
+
+mod arg;
+mod client;
+mod error;
+mod util;
+
+const TEST_SUITS: &str = "tests/logictest/suites";
+
+pub struct Databend {
+    mysql_client: Option<MysqlClient>,
+    http_client: Option<HttpClient>,
+    ck_client: Option<ClickhouseHttpClient>,
+}
+
+impl Databend {
+    pub fn create(
+        mysql_client: Option<MysqlClient>,
+        http_client: Option<HttpClient>,
+        ck_client: Option<ClickhouseHttpClient>,
+    ) -> Self {
+        Databend {
+            mysql_client,
+            http_client,
+            ck_client,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl sqllogictest::AsyncDB for Databend {
+    type Error = DSqlLogicTestError;
+
+    async fn run(&mut self, sql: &str) -> Result<DBOutput> {
+        if let Some(mysql_client) = &mut self.mysql_client {
+            println!("Running sql with mysql client: [{}]", sql);
+            return mysql_client.query(sql);
+        }
+        if let Some(http_client) = &mut self.http_client {
+            println!("Running sql with http client: [{}]", sql);
+            return http_client.query(sql).await;
+        }
+        println!("Running sql with clickhouse client: [{}]", sql);
+        self.ck_client.as_mut().unwrap().query(sql).await
```

Comment:
Why we don't need to check this `ck_client`?

Reply:
If we must chose on from those three engines, I prefer to use an `enum` here for better reading.